packages/context/src/reflect.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/authenticate
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject, Constructor} from '..';
+import 'reflect-metadata';
+
+/*
+ * namespaced wrapper to handle reflect api
+ */
+export class NamespacedReflect {
+  /**
+   * @param namespace : namespace to bind this reflect context
+   */
+  constructor(private namespace?: string) {}
+
+  /**
+   * lookup metadata from a target object and its prototype chain
+   */
+  getMetadata(metadataKey: string, target: Function, propertyKey?: string | symbol): Object {
+    // prefix namespace, if provided, to the metadata key
+    metadataKey = this.namespace? this.namespace + ':' + metadataKey : metadataKey;
```

Comment:
The implementation of prefixing algorithm is duplicated too much to my taste. Have you considered extracting it into a shared method?

```ts
export class NamespacedReflect {
  // ...
  getMetadata(metadataKey: string, target: Function, propertyKey?: string | symbol): Object {
    metadataKey = this.buildKey(metadataKey);
    // ...
  }

  buildKey(metadataKey: string) {
    return this.namespace? this.namespace + ':' + metadataKey : metadataKey;
  }
}
```